hell/adventcalendar/2024/14/index.md

```diff
@@ -12,147 +12,185 @@ active: true
+- The `focusable="false"` attribute on SVG elements prevents them from being focusable via keyboard navigation in certain browsers (e.g. Internet Explorer 11). For images, this attribute is not required, because they are automatically excluded from keyboard navigation.
 
-**Skipping Multiple Pages**:
+**Using Images**
+- When an `<img>` element is the only child of a link, the text of the `alt` attribute is automatically read by screenreaders as the link text. Therefore, the `alt` attribute should include a descriptive text that clearly conveys the function of the link (e.g. `alt="Next Page, Page 9"`). This method is a best practice as it is natively supported by screenreaders and requires no additional effort.
+- If it is not possible to modify the `alt` attribute of the image due to technical constraints, an `aria-label` on the `<a>` element can be used instead to describe the function of the link. However, this solution should only be applied in exceptional cases.
 
-* Screen readers in reading mode will also announce `<span>` elements. The `aria-label` (e.g., "skipping Pages 2 to 6") provides information on the skipped pages, as the simple "…" symbol alone would not be meaningful enough for screen readers.
-<!-- MM: aria-disabled is not supposed be used with generic elements: https://www.w3.org/TR/wai-aria/#aria-disabled -->
-<!-- MM: aria-label is not allowed on generic elements: https://www.w3.org/TR/html-aria/#docconformance-naming -->
-* Instead of requiring users to go through unnecessary page numbers, `aria-disabled` groups the inactive, skipped pages as a single unit, keeping the navigation structure clear and efficient.
-<!-- MM: NVDA ignores the span entirely and just announces an empty bullet -->
+**Skipping Multiple Pages**
+- In reading mode, screenreaders typically do not announce ellipses (“...”) used to indicate skipped pages. Instead, the list item is often read as an empty bullet, which can confuse users as it does not explain why no additional information is provided at that point.
+- To ensure that skipping pages is clearly communicated, an explanatory text such as “skipping pages 10 to 12” should be added using the [screenreader-only class .sr-only](https://webaim.org/techniques/css/invisiblecontent/). This text will be read by screenreaders but remain invisible to sighted users. An `aria-label` is not a suitable alternative here, as it cannot be applied to generic elements like `<span>` or `<div>`.
+- This approach is particularly helpful when page skipping occurs in the middle of pagination, as it provides screenreader users with context and prevents confusion about why navigation jumps from one page (e.g. page 2) to another (e.g. page 7). The additional context improves orientation and reduces potential misunderstandings.
 
 ## The final chapter
```

Comment:
Did you test your final solutions with different screen readers? If yes, which ones? You're giving a lot of advice here and I just want to make sure that you code delivers what it promises.

Reply:
Thank you for your feedback! I completely understand the importance of verifying that the solutions I propose are thoroughly tested. To address this, I’d like to include a section detailing the test environments I used to ensure the accessibility of the solutions. 

How about the following phrasing to close my post?

> ## Testing and validation
> To ensure the proposed solutions deliver on their promise, I conducted tests in different environments:
> - HTML Validation: The code was tested using the [W3C HTML Validator](https://validator.w3.org/nu/) with no errors or warnings found.
> - Tested on [AssistivLabs](https://assistivlabs.com/), a platform for remote testing of assistive technologies:
> 	- Screenreader: NVDA (Version 2024.4)
> 	- Browser: Chrome (Version 130)
> - Tested on a real device:
> 	- Operating System: macOS 15.1.1
> 	- Screenreader: VoiceOver
> 	- Browsers: Chrome (Version 131) and Firefox (Version 133)
> 
> These tests confirmed the functionality and accessibility of the solutions. However, additional testing with other screenreader and browser combinations is recommended to ensure compatibility across a wider range of devices and tools.